server/src/main/java/com/google/coffeehouse/servlets/CreatePersonServlet.java

```diff
@@ -19,24 +19,31 @@
 import com.google.coffeehouse.util.UuidWrapper;
 import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** 
- * Servlet to create a {@link Person} from URL parameter input, 
+ * Servlet to create a {@link Person} from Http POST Request Body (in JSON format),
  * save it in database, and return it in JSON format.
  */
 @WebServlet("/create-person")
 public class CreatePersonServlet extends HttpServlet {
 
   /** 
-   * The error string sent by the response object in doPost when no email or nickname parameter 
-   * exists.
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request cannot be used to construct a {@link Person} for any reason.
    */
-  public static final String EMAIL_OR_NICKNAME_ERROR = "- email or nickname not specified.";
+  public static final String BODY_ERROR = "- unable to parse body.";
+
+  /** The logged error string when an error parsing the body of the POST request is encountered */
+  public static final String LOG_BODY_ERROR_MESSAGE = 
+      "LOGGING: Body unable to be parsed in CreatePersonServlet: ";
   private IdentifierGenerator idGen = null;
+  private static final Gson gson = new Gson();
```

Comment:
Making gson a static final was a good idea for efficiency here so that we're not instantiating it each time. Nice.